### PR TITLE
TCFv2 0% Server-side Test

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Implements TCFv2 0% test in frontend.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No, **not yet**
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

The CMP is working properly, and the AB test is running smoothly.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [X] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [X] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [X] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by ripecosta — CHANGES_REQUESTED** [review]

File: static/src/javascripts/projects/commercial/modules/cmp/cmp.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 // @flow strict
 
+import { onConsentChange, oldCmp } from '@guardian/consent-management-platform';
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/cmp/cmp.js]: That make sense. Changes have been made to reflect this!

File: common/app/dev/DevParametersHttpRequestHandler.scala [hunk 2]
```diff
@@ -48,8 +48,8 @@ class DevParametersHttpRequestHandler(
     "heatmap", // used by ophan javascript to enable the heatmap
     "format", // used to determine whether HTML should be served in email-friendly format or not
     "timestamp", //used to get specific builds for inteactive serviceworkers
-    "pbjs_debug", // set to true to enable prebid debugging,
-    "amzn_debug_mode" // set to 1 to enable A9 debugging
+    "pbjs_debug", // set to `true` to enable prebid debugging,
```
Comment (ripecosta) [inline common/app/dev/DevParametersHttpRequestHandler.scala:51]: Was this intended?

File: common/app/dev/DevParametersHttpRequestHandler.scala [hunk 3]
```diff
@@ -48,8 +48,8 @@ class DevParametersHttpRequestHandler(
...
+    "amzn_debug_mode", // set to `1` to enable A9 debugging
```
Comment (ripecosta) [inline common/app/dev/DevParametersHttpRequestHandler.scala:52]: Were these changes intended?

File: static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.js [hunk 4]
```diff
@@ -9,7 +9,13 @@ import type {
     HeaderBiddingSize,
     HeaderBiddingSlot,
 } from 'commercial/modules/header-bidding/types';
-import { onIabConsentNotification } from '@guardian/consent-management-platform';
+
+import { onConsentChange, oldCmp } from '@guardian/consent-management-platform';
+import { isInTcfv2Test } from 'commercial/modules/cmp/tcfv2-test';
+
+const onIabConsentNotification = isInTcfv2Test()
```
Comment (ripecosta) [inline static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.js:16]: Did you forget to call it `onCMPConsentNotification` here like in the other places?
I think you also meant to update the callback that gets passed into `onIabConsentNotification` to be able to account for the TCFv2 state objects on line 42 of this file.

File: static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.spec.js [hunk 5]
```diff
@@ -1,9 +1,10 @@
 // @flow
 
 import a9, { _ } from 'commercial/modules/header-bidding/a9/a9';
-import { onIabConsentNotification as onIabConsentNotification_ } from '@guardian/consent-management-platform';
+import {
+    oldCmp as oldCmp_ } from '@guardian/consent-management-platform';
```
Comment (ripecosta) [inline static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.spec.js:5]: Was this intentional?

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js [hunk 6]
```diff
@@ -14,10 +14,24 @@ import { permutive } from 'commercial/modules/third-party-tags/permutive';
...
+// TODO: Remove oldCmp.onGuConsentNotification and justuse onConsentChange when tcfv2 is released
+const onGuConsentNotification = isInTcfv2Test()
+    ? (fake, callback) => {
+        onConsentChange(callback);
+        console.warn(
+            'the onGuConsentNotification method is deprecated',
+            fake
+        );
+    }
+    : oldCmp.onGuConsentNotification;
```
Comment (ripecosta) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js:34]: I believe this can be removed. You're not using it.

File: static/src/javascripts/projects/common/modules/atoms/youtube-player.js [hunk 7]
```diff
@@ -6,11 +6,18 @@ import { loadScript } from 'lib/load-script';
 import { constructQuery } from 'lib/url';
 import { getPageTargeting } from 'common/modules/commercial/build-page-targeting';
 import { commercialFeatures } from 'common/modules/commercial/commercial-features';
-import { onIabConsentNotification } from '@guardian/consent-management-platform';
 import $ from 'lib/$';
 import { buildPfpEvent } from 'common/modules/video/ga-helper';
+
```
Comment (ripecosta) [inline static/src/javascripts/projects/common/modules/atoms/youtube-player.js:11]: Just being picky here but I would prefer to avoid leaving blank lines between imports. Same on line 14.

File: static/src/javascripts/projects/common/modules/commercial/build-page-targeting.spec.js [hunk 8]
```diff
@@ -16,22 +16,28 @@ import { getSync as getSync_ } from 'lib/geolocation';
 import { isUserLoggedIn as isUserLoggedIn_ } from 'common/modules/identity/api';
 import { getUserSegments as getUserSegments_ } from 'common/modules/commercial/user-ad-targeting';
 import { getSynchronousParticipations as getSynchronousParticipations_ } from 'common/modules/experiments/ab';
-import { onIabConsentNotification as onIabConsentNotification_ } from '@guardian/consent-management-platform';
+import {
```
Comment (ripecosta) [inline static/src/javascripts/projects/common/modules/commercial/build-page-targeting.spec.js:19]: Was this intentional?

**Review by ripecosta — CHANGES_REQUESTED** [review]
Requesting changes just because I believe you forgot something the a9 file.

File: common/app/dev/DevParametersHttpRequestHandler.scala (hunk 3 above)
Comment (mxdvl) [reply common/app/dev/DevParametersHttpRequestHandler.scala:52]: It was clearer in my view, but happy to revert.

```suggestion
    "pbjs_debug", // set to true to enable prebid debugging,
    "amzn_debug_mode", // set to 1 to enable A9 debugging
```

File: static/src/javascripts/projects/common/modules/atoms/youtube-player.js (hunk 7 above)
Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/atoms/youtube-player.js:11]:
```suggestion
```

File: common/app/dev/DevParametersHttpRequestHandler.scala (hunk 3 above)
Comment (ripecosta) [reply common/app/dev/DevParametersHttpRequestHandler.scala:52]: Not a problem. It seemed like the linter was trying to lint the comments which I found odd. Resolving this.

File: static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.spec.js (hunk 5 above)
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.spec.js:5]: It should fit on a single line.

**Review by ripecosta — APPROVED** [review]
Amazing! Great work! 👍

**Comment by mxdvl** [comment]
Only a few bug fixes needed, exposing `setErrorHandler` function from the oldCMP and fixing the IAB stub.

**Comment by mxdvl** [comment]
Once consent functionality is implemented and the 1% test has run, DCR needs to have TCFv2 implemented as per the following PR: https://github.com/guardian/dotcom-rendering/pull/1729
